main.nf

```diff
@@ -3,219 +3,277 @@
-		--normalisation-method ${params.download_data.normalisation_method}\
-		--output-dir-name ${params.download_data.query_output_dir}\
-		--get-condensed-sdrf\
-		--get-marker-genes\
-		--number-of-clusters ${params.download_data.query_num_clust}
-	# rename files to avoid name collisions in subsequent processes
-	mv ${query_data}/10x_data ${query_data}/query_10x_data
-	mv ${query_data}/condensed-sdrf.tsv ${query_data}/query_sdrf.txt
-	mv ${query_data}/${query_markers} ${query_data}/query_${query_markers}
+	if [ ${seq_method} ==  "droplet" ]; then
+        MATRIX_TYPE_UPD="CPM"
+    	else
+        MATRIX_TYPE_UPD=${matrix_type}
+    	fi
+    	get_experiment_data.R\
+                --accesssion-code ${dataset_id}\
+                --config-file ${params.data_import.scxa_import_config_file}\
+             	--matrix-type CPM\
+                --output-dir-name ${dataset_id}\
+                --get-sdrf ${params.data_import.get_sdrf}\
+                --get-condensed-sdrf ${params.data_import.get_cond_sdrf}\
+                --get-idf ${params.data_import.get_idf}\
+                --get-marker-genes ${params.data_import.get_marker_genes}\
+                --number-of-clusters ${num_clust}
 	"""
     }
-  
-	DATASET_ID = Channel.from(params.download_data.query_acc_code).first() 
 }else{
-	QUERY_10X_DIR = Channel.fromPath(params.data_10X, checkIfExists: true).first()
-	CONDENSED_SDRF = Channel.fromPath(params.condensed_sdrf, checkIfExists: true).first()
-	MARKERS = Channel.fromPath(params.markers_metadata, checkIfExists: true).first()
-	DATASET_ID = Channel.from(params.dataset_id) 
+	//QUERY_10X_DIR = Channel.fromPath(params.data_10X, checkIfExists: true).first()
+	//CONDENSED_SDRF = Channel.fromPath(params.condensed_sdrf, checkIfExists: true).first()
+	//MARKERS = Channel.fromPath(params.markers_metadata, checkIfExists: true).first()
+	//DATASET_ID = Channel.from(params.dataset_id) 
 }	
 
+
 // condensed sdrf files need 'un-melting' 
 process unmelt_sdrf_query {
+	//publishDir "${params.data_dir}", mode: 'copy'
 	conda "${baseDir}/envs/exp_metadata.yaml"
 	
 	input:
-	file(condensed_sdrf) from CONDENSED_SDRF
+	tuple file(data), val(dataset_id), val(barcode_col), val(cell_type_col), val(matrix_type), val(num_clust) from FETCH_DATA
 	output:
-	file("query_metadata.tsv") into UNMELTED_SDRF
+        tuple file("${data}.${matrix_type}"), val(dataset_id), val(barcode_col), val(cell_type_col), val(matrix_type), val(num_clust) into FETCH_DATA_UNMELT
 	"""
 	unmelt_condensed.R\
-		-i ${condensed_sdrf}\
-		-o query_metadata.tsv\
+		-i ${data}/condensed-sdrf.tsv\
+		-o ${data}/unmelted_sdrf.tsv\
 		--retain-types ${params.unmelt_sdrf.retain_types}\
 		--has-ontology                 
+	# rename data dir name to avoid downstream file name collision
+	mv ${data} ${data}.${matrix_type} 
 	"""
 	}
 
+// duplicate channel 
+//FETCH_DATA_UNMELT.into{ BARCODES; CPM_DATA; TPM_DATA }
+FETCH_DATA_UNMELT.into{ BARCODES; DATA_UNMELT }
```

Comment:
@rubenchazarra I didn't quite understand what's happening here. Could you explain?

Reply:
Hi @a-solovyev12, here I'm duplicating the channel so I can pass the barcodes to the `generate_folds` process, and later on pass the same channel with the data to the `split_train_test` process